### PR TITLE
Move reusable logic between Transcoder and GFX Filter to util

### PR DESCRIPTION
**Pull request by dchakarwarti**

### CODE REVIEW

**Comment by dchakarwarti** [comment]
cc: @nareddyt @TAOXUY

File: src/go/configgenerator/filtergen/util_test.go [hunk 1]
```diff
@@ -1,15 +1,57 @@
...
 import (
+	"fmt"
 	"testing"
 
 	"github.com/GoogleCloudPlatform/esp-v2/src/go/options"
 	"github.com/google/go-cmp/cmp"
 	servicepb "google.golang.org/genproto/googleapis/api/serviceconfig"
 	apipb "google.golang.org/genproto/protobuf/api"
 	"google.golang.org/protobuf/testing/protocmp"
+
```
Comment (nareddyt) [inline src/go/configgenerator/filtergen/util_test.go:12]: nit: fix formatting of imports, there should be no blank lines

File: src/go/configgenerator/filtergen/util_test.go [hunk 2]
```diff
@@ -379,3 +421,123 @@ func TestGetAPIKeySystemParametersBySelectorFromOPConfig(t *testing.T) {
...
+	invalidRawDescriptor := []byte("invalid_random_descriptor")
+	validContentwithInvalidProto, err := setupSourceFileContent(invalidRawDescriptor)
+	if err != nil {
+		t.Fatalf("Failed to setup source file content: %v", err)
+	}
+
+	invalidContentwithInvalidProto := setupInvalidSourceFileContent(invalidRawDescriptor)
+
+	testdata := []struct {
+		Desc            string
```
Comment (nareddyt) [inline src/go/configgenerator/filtergen/util_test.go:462]: lowercase vars, they should not be exported

File: src/go/configgenerator/filtergen/util_test.go [hunk 3]
```diff
@@ -379,3 +421,123 @@ func TestGetAPIKeySystemParametersBySelectorFromOPConfig(t *testing.T) {
...
+				},
+			},
+			DescriptorBin: rawDescriptor,
+			WantError:     false,
+		},
+		{
+			Desc: "Failure in extracting descriptor bytes from the OP service config. Valid content with invalid proto.",
+			ServiceConfigIn: &servicepb.Service{
+				Name: "bookstore.endpoints.project123.cloud.goog",
+				Apis: []*apipb.Api{
```
Comment (nareddyt) [inline src/go/configgenerator/filtergen/util_test.go:492]: No need for Apis section, remove anything irrelevant to the test

File: src/go/configgenerator/filtergen/util_test.go [hunk 4]
```diff
@@ -379,3 +421,123 @@ func TestGetAPIKeySystemParametersBySelectorFromOPConfig(t *testing.T) {
...
+			WantError: true,
+		},
+	}
+	for _, tc := range testdata {
+		t.Run(tc.Desc, func(t *testing.T) {
+			descriptorBin, err := GetDescriptorBinFromOPConfig(tc.ServiceConfigIn)
+			if tc.WantError && err == nil {
+				t.Fatalf("GetDescriptorBinFromOPConfig() got error: %v, want error presence: %v", err, tc.WantError)
+			}
+			if !tc.WantError && err != nil {
```
Comment (nareddyt) [inline src/go/configgenerator/filtergen/util_test.go:535]: suggest you break this into two separate tests, one for positive flow, one for error flow. The test functions will be a lot simpler.

[go/gotip/episodes/50](https://goto.google.com/gotip/episodes/50)

**Review by nareddyt — COMMENTED** [review]
Thanks for doing this, looks great!

**Review by TAOXUY — APPROVED** [review]

**Comment by dchakarwarti** [comment]
Thanks for the review. Addressed comments.

**Review by nareddyt — APPROVED** [review]

**Pull request merged**